### PR TITLE
Remove some const-away casts from DQM and validation

### PR DESCRIPTION
**Pull request by makortel**
To clean up static analyzer reports.

Tested in CMSSW_10_4_X_2018-10-25-2300, no changes expected.

### CODE REVIEW

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by andrius-k** [comment]
+1

File: DQM/EcalMonitorTasks/src/RawDataTask.cc [hunk 1]
```diff
@@ -61,7 +61,7 @@ namespace ecaldqm {
     // Get GT L1 info
     const FEDRawData &gtFED(_fedRaw.FEDData(812));
     if(gtFED.size() > sizeof(uint64_t)){ // FED header is one 64 bit word
-      uint32_t *halfHeader((uint32_t *)gtFED.data());
+      const uint32_t *halfHeader = reinterpret_cast<const uint32_t *>(gtFED.data());
```
Comment (fabiocos) [inline DQM/EcalMonitorTasks/src/RawDataTask.cc]: @makortel for my understanding, could you please explain why here reinterpret_cast is preferable to static_cast?

**Comment by makortel** [comment]
Took the opportunity to remove all (unnecessary) `const_cast`s from `DQM/SiStripCommissioningSources`.

File: DQM/EcalMonitorTasks/src/RawDataTask.cc (hunk 1 above)
Comment (makortel) [reply DQM/EcalMonitorTasks/src/RawDataTask.cc]: @fabiocos 
The return type of `FEDRawData::data()` is `const unsigned char *`, which then gets re-interpreted here as `const uint32_t *`.

**Comment by makortel** [comment]
@cmsbuild, please test

File: DQM/EcalMonitorTasks/src/RawDataTask.cc (hunk 1 above)
Comment (fabiocos) [reply DQM/EcalMonitorTasks/src/RawDataTask.cc]: ok, there is no relationship among the original and casted version...

**Comment by andrius-k** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**